pkg/maps/ctmap/ctmap.go

```diff
@@ -360,24 +361,38 @@ func (f *GCFilter) doFiltering(dstIP net.IP, dstPort uint16, nextHdr, flags uint
 		action = deleteEntry
 	}
 
-	// used by FlushCTEntriesOf
-	// Delete all entries of the given dstIP that are not filtered
-	if f.Type&GCFilterByIDsToKeep != 0 &&
-		dstIP.Equal(f.IP) &&
-		// only delete TUPLE_F_IN connections since the clean up is made
-		// for ingress policies
-		flags&TUPLE_F_IN != 0 {
+	// If the filter doesn't contain an endpoint ID & IP, no entries will get matched below.
+	if f.EndpointID == 0 || f.EndpointIP == nil {
```

Comment:
Type definition comments above let me believe that you could give one of the (IP, ID) as zero, but here you return even if one of them is non-zero. Maybe update the comments in the definition?

Reply:
Done.